src/runtime_src/core/pcie/driver/linux/xocl/subdev/hwmon_sdm.c

```diff
@@ -833,7 +833,10 @@ mac_addr0_show(struct device *dev, struct device_attribute *attr, char *buf)
 {
 	struct xocl_hwmon_sdm *sdm = dev_get_drvdata(dev);
 
-	return sprintf(buf, "0x%llx\n", sdm->bdinfo.mac_addr0);
+	return sprintf(buf, "%02hhX:%02hhX:%02hhX:%02hhX:%02hhX:%02hhX:%02hhX:%02hhX\n",
+				   sdm->bdinfo.mac_addr0[0], sdm->bdinfo.mac_addr0[1], sdm->bdinfo.mac_addr0[2],
+				   sdm->bdinfo.mac_addr0[3], sdm->bdinfo.mac_addr0[4], sdm->bdinfo.mac_addr0[5],
+				   sdm->bdinfo.mac_addr0[6], sdm->bdinfo.mac_addr0[7]);
```

Comment:
Is there a better way to show mac_addr0[0:7]?

Reply:
If I am using %X:%X:* then seeing overflow values like below. 

-bash-4.2# cat /sys/bus/pci/devices/0000\:d8\:00.0/hwmon_sdm.m.55574528/mac_addr0 -> printing type is %02X
00:0A:35:0C:FFFFFFF6:3C:00:FFFFFFA4

So, I had to use %02hhX. Are you asking this only or something else?